src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -404,6 +407,9 @@ private void deletePod(String id) {
 			boolean podsDeleted = podsToDelete.delete();
 			logger.debug(String.format("Pod deleted for: %s - %b", id, podsDeleted));
 		}
+		else {
+			throw new IllegalStateException(String.format("Cannot delete Pod for task: %s. Because Pod is not exist!", id));
```

Comment:
I think we can replace the "!" with a reason phrase at the end of the message wrapped in parens to make it stand out. Something like:
```java
throw new IllegalStateException(String.format("Cannot delete Pod for task \"%s\" (reason: pod does not exist)", id));
```

The output before:
```
Cannot delete Pod for task: foo. Because Pod is not exist!
```
The output after: 
```
Cannot delete Pod for task "foo" (reason: pod does not exist)
```
Wdyt @umutcann ?